core/src/main/java/org/apache/accumulo/core/metadata/AbstractTabletFile.java

```diff
@@ -62,4 +62,17 @@ public Range getRange() {
   public boolean hasRange() {
     return !range.isInfiniteStartKey() || !range.isInfiniteStopKey();
   }
+
+  /**
+   * Return the exact string (full Json with range) that is stored in the metadata table.
+   */
+  public abstract String getMetadata();
```

Comment:
Why were these methods relating to what is in the metadata table added here instead of just in StoredTabletFile?

Reply:
I had added it as ReferencedTabletFile needs metadata for inserts but it doesn't really make sense to put it there as shown by the Unsupported operations in UnreferencedTabletFile. I will remove it and if ReferencedTabletFile is doing an insert it will just create a StoredTabletFile and delegate like it has before.